examples/jetstream/push-consumer/java/Main.java

```diff
@@ -0,0 +1,198 @@
+package example;
+
+import io.nats.client.*;
+import io.nats.client.api.*;
+import io.nats.client.impl.NatsJetStreamMetaData;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.concurrent.CountDownLatch;
+
+public class Main {
+    public static void main(String[] args) {
+        String natsURL = System.getenv("NATS_URL");
+        if (natsURL == null) {
+            natsURL = "nats://127.0.0.1:4222";
+        }
+
+        // Initialize a connection to the server. The connection is AutoCloseable
+        // on exit.
+        try (Connection nc = Nats.connect(natsURL)) {
+
+            // Access `JetStream` and `JetStreamManagement` which provide methods to create
+            // streams and consumers as well as convenience methods for publishing
+            // to streams and consuming messages from the streams.
+            JetStream js = nc.jetStream();
+            JetStreamManagement jsm = nc.jetStreamManagement();
+
+            // Declare a simple [limits-based stream][1] and populate the stream
+            // with a few messages.
+            // [1]: /examples/jetstream/limits-stream/java/
+            String streamName = "EVENTS";
+            StreamConfiguration config = StreamConfiguration.builder()
+                    .name(streamName)
+                    .subjects("events.>")
+                    .build();
+
+            StreamInfo stream = jsm.addStream(config);
+
+            js.publish("events.1", null);
+            js.publish("events.2", null);
+            js.publish("events.3", null);
+
+            // ### Ephemeral
+            // The JetStream context provides a simple way to create an ephemeral
+            // push consumer, simply provide a subject that overlaps with the
+            // bound subjects on the stream and this helper method will do the
+            // stream look-up automatically and create the consumer.
```

Comment:
It does not have to match bound, it must match the stream subject somehow, the server has a pile of code to figure this out. If you created the consumer as you do here, but did it before publishing, the consumer would be created.

```
StreamConfiguration config = StreamConfiguration.builder()
    .name("EVENTS")
    .subjects("events.>")
    .build();

jsm.addStream(config);
JetStreamSubscription sub = js.subscribe("events.foo");
```

Reply:
Currently the doc mentions 'subject that overlaps with the bound subjects'. Indeed it does not need to match since it doesn't need to be equal, but it can have overlap? Where `events.foo` has overlap with the stream-defined `events.>`.

Is the current doc sufficient or would this need to be extended beyond just  'overlaps'?

Reply:
I now understand it does need to match and does NOT check for overlap.
Updated the docs to be `..., simply provide a subject that matches with one of the bound subjects on the stream..'`